packages/ckeditor5-table/src/converters/upcasttable.ts

```diff
@@ -263,6 +275,11 @@ function scanTable( viewTable: ViewElement ) {
 					headingColumns = headingCols;
 				}
 			}
+
+			// We use the maximum number of columns to avoid false positives when detecting
+			// multiple rows with single column within `rowspan`. Without it the last row of `rowspan=3`
+			// would be detected as a heading row because it has only one column (identical to the previous row).
+			maxPrevColumns = Math.max( maxPrevColumns || -Infinity, trColumns.length );
```

Comment:
Can we make it simpler like this?

```suggestion
			maxPrevColumns = Math.max( maxPrevColumns || 0, trColumns.length );
```